source/sphinx_extensions/command_block/extension.py

```diff
@@ -40,17 +63,34 @@ class CommandBlockDirective(docutils.parsers.rst.Directive):
     has_content = True
 
     option_spec = {
-        'no-exec': docutils.parsers.rst.directives.flag
+        'no-exec': docutils.parsers.rst.directives.flag,
+        'url': docutils.parsers.rst.directives.unchanged_required,
+        'rename': docutils.parsers.rst.directives.unchanged_required,
     }
 
     def run(self):
-        self.assert_has_content()
-        commands = functools.reduce(self._parse_multiline_commands,
-                                    self.content, [])
-
-        nodes = [
-            self._get_literal_block_node(self.content)
-        ]
+        command_mode = True if self.name == 'command-block' else False
+        opts = self.options
+        download_opts = [k in opts for k in ['url', 'rename']]
+
+        if command_mode:
+            self.assert_has_content()
+            if any(download_opts):
+                raise self.error('command-block does not support the '
```

Comment:
Can you raise `sphinx.errors.ExtensionError` instead? That's what's used for error raising in other parts of this extension, and it's the recommended way to raise an error within an extension (see the note at the start of [this section](http://www.sphinx-doc.org/en/1.5.2/extdev/appapi.html#producing-messages-logging) of the docs).